source/win-spout-filter.cpp

```diff
@@ -104,76 +94,82 @@ void win_spout_offscreen_render(void* data, uint32_t cx, uint32_t cy)
 	uint32_t width = obs_source_get_base_width(target);
 	uint32_t height = obs_source_get_base_height(target);
 
-	gs_texrender_reset(context->texrender);
-
-	if (gs_texrender_begin(context->texrender, width, height))
-	{
+	// Render the target to an intemediate format in sRGB-aware format
+	gs_texrender_reset(context->texrender_intermediate);
+	if (gs_texrender_begin(context->texrender_intermediate, width, height)) {
 		struct vec4 background;
 		vec4_zero(&background);
 
 		gs_clear(GS_CLEAR_COLOR, &background, 0.0f, 0);
-		gs_ortho(0.0f, (float)width, 0.0f, (float)height, -100.0f, 100.0f);
+		gs_ortho(0.0f, (float)width, 0.0f, (float)height, -100.0f,
+			 100.0f);
 
 		gs_blend_state_push();
 		gs_blend_function(GS_BLEND_ONE, GS_BLEND_ZERO);
 
 		obs_source_video_render(target);
 
 		gs_blend_state_pop();
-		gs_texrender_end(context->texrender);
+		gs_texrender_end(context->texrender_intermediate);
+	}
 
-		if (context->width != width || context->height != height)
-		{
+	// Use the default effect to render it back into a format Spout accepts
+	gs_texrender_reset(context->texrender_curr);
+	if (gs_texrender_begin(context->texrender_curr, width, height))
+	{
+		struct vec4 background;
+		vec4_zero(&background);
 
-			gs_stagesurface_destroy(context->stagesurface);
-			context->stagesurface = gs_stagesurface_create(width, height, GS_BGRA);
+		gs_clear(GS_CLEAR_COLOR, &background, 0.0f, 0);
+		gs_ortho(0.0f, (float)width, 0.0f, (float)height, -100.0f, 100.0f);
+
+		gs_blend_state_push();
+		gs_blend_function(GS_BLEND_ONE, GS_BLEND_ZERO);
+
+		// const bool previous = gs_set_linear_srgb(false);
 
-			video_output_info video_out = { 0 };
-			video_out.format = VIDEO_FORMAT_BGRA;
-			video_out.width = width;
-			video_out.height = height;
-			video_out.fps_den = context->video_info.fps_den;
-			video_out.fps_num = context->video_info.fps_num;
-			video_out.cache_size = 16;
-			video_out.colorspace = VIDEO_CS_DEFAULT;
-			video_out.range = VIDEO_RANGE_DEFAULT;
-			video_out.name = obs_source_get_name(context->source_context);
+		// obs_source_video_render(target);
```

Comment:
Do you think we could remove these commented lines if they are not being used any more?